lib/capability/capability_test.go

```diff
@@ -0,0 +1,275 @@
+package capability
+
+import (
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+func TestSetCapabilities(t *testing.T) {
+	tests := []struct {
+		name            string
+		config          *configv1.ClusterVersion
+		wantKnownKeys   []string
+		wantEnabledKeys []string
+	}{
+		{name: "capabilities nil",
+			config:          &configv1.ClusterVersion{},
+			wantKnownKeys:   []string{string(configv1.ClusterVersionCapabilityOpenShiftSamples)},
+			wantEnabledKeys: []string{string(configv1.ClusterVersionCapabilityOpenShiftSamples)},
+		},
+		{name: "capabilities set not set",
+			config: &configv1.ClusterVersion{
+				Spec: configv1.ClusterVersionSpec{
+					Capabilities: &configv1.ClusterVersionCapabilitiesSpec{},
+				},
+			},
+			wantKnownKeys:   []string{string(configv1.ClusterVersionCapabilityOpenShiftSamples)},
+			wantEnabledKeys: []string{string(configv1.ClusterVersionCapabilityOpenShiftSamples)},
+		},
+		{name: "set capabilities None",
+			config: &configv1.ClusterVersion{
+				Spec: configv1.ClusterVersionSpec{
+					Capabilities: &configv1.ClusterVersionCapabilitiesSpec{
+						BaselineCapabilitySet:         configv1.ClusterVersionCapabilitySetNone,
+						AdditionalEnabledCapabilities: []configv1.ClusterVersionCapability{},
```

Comment:
can we drop this line?  I expect most folks who use `None` to not also include an explicit empty array for additional caps.